src/aix.rs

```diff
@@ -0,0 +1,31 @@
+use std::num::NonZeroUsize;
+use std::io::{BufReader, Read, Seek, SeekFrom};
+use std::fs::File;
+
+extern "C" {
+    fn getpid() -> i32;
+}
+
+// https://www.ibm.com/docs/en/aix/7.3?topic=files-proc-file
+//
+// The 'status' file is defined as binary format below
+//   uint32_t pr_flag;  /* process flags from proc struct p_flag */
+//   uint32_t pr_flag2; /* process flags from proc struct p_flag2 */
+//   uint32_t pr_flags; /* /proc flags */
+//   uint32_t pr_nlwp;  /* number of threads in the process */
+//   ...                /* Omitted */
+pub(crate) fn num_threads() -> Option<NonZeroUsize> {
+    let pid = unsafe {
+        getpid()
+    };
+    let file = File::open(format!("/proc/{}/status", pid)).ok()?;
```

Comment:
I don't suppose Aix has a "current" path so you can retrieve information for the current process from a static location?

Even better would be something like sysctl() to retrieve it without relying on the filesystem.

Reply:
Unfornately, AIX does not have `/proc/self` or such system call for 'current' process.